pkg/cvo/upgradeable.go

```diff
@@ -152,6 +153,31 @@ func (check *clusterOperatorsUpgradeable) Check() *configv1.ClusterOperatorStatu
 		return cond
 	}
 
+	// Set upgradeable = false to stop the y stream upgrades to happen when an upgrade is in progress.
```

Comment:
Seems like this logic should be moved up into syncUpgradeable and this Check method should simply continue to return cond(s) to be handled there. syncUpgradeable already has access to `optr`, calls `setUpgradeable` and does the requeue.

Reply:
syncUpgradeable is called from a different go routine and it is only supposed to be synchronizing the upgradeable false conditions. But in this case we want to set the upgradeable = false and then syncUpgradeable will do it jobs.

Reply:
'Check` is called by `syncUpgradeable` here https://github.com/openshift/cluster-version-operator/blob/478e60d7efed233d9979c70fcaf8c097b2d5d6f5/pkg/cvo/upgradeable.go#L36. Is there another call to `Check` that you plan to add on one you have added that I missed?